From 4b0aa7fe07ecd49d73b477525d9f3a6e39bf1af2 Mon Sep 17 00:00:00 2001
From: Tomasz Duszynski <tduszynski@cavium.com>
Date: Wed, 13 Mar 2019 16:34:00 +0100
Subject: [PATCH 045/386] octeontx2-af: fix warning by removing unnecessary
 comparison
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

crypto_eng_grp is defined as unsigned hence comparison can be removed.
Doing so we also get rid of following warning:

drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c: In function ‘rvu_mbox_handler_cpt_set_crypto_grp’:
drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c:300:27: warning: comparison is always false due to limited range of data type [-Wtype-limits]
      (req->crypto_eng_grp < 0 ||
                           ^

Change-Id: I619275edabf32d0a050f0e18f8ffd8e573f10fd1
Signed-off-by: Tomasz Duszynski <tduszynski@marvell.com>
Acked-by: Lukasz Bartosik <lbartosik@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5718
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
index acaf1ce340c2..99dd130e2895 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
@@ -297,8 +297,7 @@ int rvu_mbox_handler_cpt_set_crypto_grp(struct rvu *rvu,
 	rsp->crypto_eng_grp = req->crypto_eng_grp;
 
 	if (req->crypto_eng_grp != INVALID_ENGINE_GRP &&
-	    (req->crypto_eng_grp < 0 ||
-	    req->crypto_eng_grp >= CPT_MAX_ENGINE_GROUPS))
+	    req->crypto_eng_grp >= CPT_MAX_ENGINE_GROUPS)
 		return CPT_AF_ERR_GRP_INVALID;
 
 	crypto_eng_grp = req->crypto_eng_grp;
-- 
2.17.1

